### PR TITLE
make mdadm raiding more resillient

### PR DESCRIPTION
**Pull request by Mpdreamz**
This is a PR with two changes that made our disk provisioning more resilient over at https://github.com/elastic/azure-marketplace

The first problem was that quite often `mdadm` was not being installed properly and the command was never available on the machine. We solved this by retrying `apt-get update` and installing mdadm under sudo.

The other thing we saw was that quite often we'd get `resource busy` messages from `mdadm` which caused `mkfs.ext4` to fail and later on in the script the call to `blkid` to not return a UUID for `/dev/md0` which in turn made the call to `add_to_fstab` to be called with no UUID.

In short this caused `/datadisks/disk1` to point to the local `sda` partition `30gb` instead of the attached raided disks (2TB in this particular case).

The cause for this seems to be a race condition and the fix documented here:

http://dev.bizo.com/2012/07/mdadm-device-or-resource-busy.html

Has worked very well on our template.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@Mpdreamz__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

**Comment by singhkay** [comment]
@Mpdreamz Thank you for the fix! merged
